### PR TITLE
Library level documentation for basic use-cases

### PR DESCRIPTION
**Pull request by pimeys**
Also allows using the SQL Browser on all platforms.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,170 @@
-//! A pure-rust TDS implementation for Microsoft SQL Server (>=2008)
+//! An asynchronous, runtime-independent, pure-rust TDS implementation for
```
Comment (tomhoule) [inline src/lib.rs:1]: I don't know if that's in scope for these docs to explain this, but I don't know what TDS stands for.

File: src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,170 @@
...
+//!     // We'll disable the Nagle algorithm. Buffering is handled
+//!     // internally with a `Sink`.
+//!     tcp.set_nodelay(true)?;
+//!
+//!     // Handling TLS, login and other details related to the SQL Server.
+//!     let mut client = Client::connect(config, tcp).await?;
+//!
+//!     // A response to a query is a stream of data, that must be
+//!     // polled empty before querying again. Using streams allows
+//!     // fetching data in an asynchronous matter, if needed.
```
Comment (tomhoule) [inline src/lib.rs:37]: typo: s/matter/manner/

File: src/lib.rs [hunk 3]
```diff
@@ -1,4 +1,170 @@
...
+//!     
+//!     // We'll disable the Nagle algorithm. Buffering is handled
+//!     // internally with a `Sink`.
+//!     tcp.set_nodelay(true)?;
+//!
+//!     // Handling TLS, login and other details related to the SQL Server.
+//!     let mut client = Client::connect(config, tcp).await?;
+//!
+//!     // A response to a query is a stream of data, that must be
+//!     // polled empty before querying again. Using streams allows
```
Comment (tomhoule) [inline src/lib.rs:36]: Nitpick: I had to read twice to understand, maybe "polled until empty", or "polled to the end"?

File: src/lib.rs (hunk 1 above)
Comment (pimeys) [reply src/lib.rs:1]: Yeah, you kind of need to understand the Microsoft terminology. Maybe I'll extend this to Tabular Data Stream (TDS).

**Pull request merged**